packages/frameworks-solid-start/src/client.tsx

```diff
@@ -0,0 +1,322 @@
+        // or if there are events from other tabs/windows
+        if (storageEvent || __NEXTAUTH._session === undefined) {
+          __NEXTAUTH._lastSync = now()
+          __NEXTAUTH._session = await getSession()
+          setSession(__NEXTAUTH._session)
+          return
+        }
+
+        if (
+          // If there is no time defined for when a session should be considered
+          // stale, then it's okay to use the value we have until an event is
+          // triggered which updates it
+          !event ||
+          // If the client doesn't have a session then we don't need to call
+          // the server to check if it does (if they have signed in via another
+          // tab or window that will come through as a "stroage" event
+          // event anyway)
+          __NEXTAUTH._session === null ||
+          // Bail out early if the client session is not stale yet
+          now() < __NEXTAUTH._lastSync
+        ) {
+          return
+        }
+
+        // An event or session staleness occurred, update the client session.
+        __NEXTAUTH._lastSync = now()
+        __NEXTAUTH._session = await getSession()
+        setSession(__NEXTAUTH._session)
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    __NEXTAUTH._getSession()
+
+    onCleanup(() => {
+      __NEXTAUTH._lastSync = 0
+      __NEXTAUTH._session = undefined
+      __NEXTAUTH._getSession = () => {}
+    })
+  })
+
+  createEffect(() => {
+    const { refetchOnWindowFocus = true } = props
+    // Listen for when the page is visible, if the user switches tabs
+    // and makes our tab visible again, re-fetch the session, but only if
+    // this feature is not disabled.
+    const visibilityHandler = () => {
+      if (refetchOnWindowFocus && document.visibilityState === "visible")
+        __NEXTAUTH._getSession({ event: "visibilitychange" })
+    }
+    document.addEventListener("visibilitychange", visibilityHandler, false)
+    onCleanup(() =>
+      document.removeEventListener("visibilitychange", visibilityHandler, false)
+    )
+  })
+
+  createEffect(() => {
+    if (refetchInterval) {
+      const refetchIntervalTimer = setInterval(() => {
+        if (__NEXTAUTH._session) {
+          __NEXTAUTH._getSession({ event: "poll" })
+        }
+      }, refetchInterval * 1000)
+      onCleanup(() => clearInterval(refetchIntervalTimer))
+    }
+  })
+
+  const value: any = createMemo(() => ({
+    data: session(),
+    status: loading()
+      ? "loading"
+      : session()
+      ? "authenticated"
+      : "unauthenticated",
+  }))
+
+  return (
+    <SessionContext.Provider value={value}>
+      {props.children}
+    </SessionContext.Provider>
+  )
+}
+
+/**
+ * Client-side method to initiate a signin flow
+ * or send the user to the signin page listing all possible providers.
+ * Automatically adds the CSRF token to the request.
+ *
+ * [Documentation](https://next-auth.js.org/getting-started/client#signin)
+ */
+export async function signIn<
```

Comment:
I barely started looking into SolidStart. but why not Actions/Forms?

https://start.solidjs.com/core-concepts/actions

Reply:
What is the point of form in this? We want people to be able to signIn from everywhere with every provider, a function is nice and can be used anywhere

Reply:
Discussed this on twitter